app/lib/audit/backend.dart

```diff
@@ -136,4 +139,68 @@ class AuditBackend {
     InvalidInputException.check(parsed != null, 'Unable to parse `before`.');
     return parsed!;
   }
+
+  /// Returns the entries from the last day.
+  ///
+  /// Keeps the [_cachedRecords] fields updated, and lists only the entries
+  /// up to the last query.
+  Future<List<AuditLogRecord>> getEntriesFromLastDay() async {
+    if (_cacheRecordsUpdateFuture != null) {
+      await _cacheRecordsUpdateFuture;
+    } else {
+      final now = clock.now().toUtc();
+      final cachedAge = now.difference(_cachedRecords.updated);
+      if (cachedAge.inSeconds < 3) {
+        return _cachedRecords.records;
+      }
+
+      // we should have only one update running
+      _cacheRecordsUpdateFuture = _updateEntriesFromLastDay(
+        cachedAge: cachedAge,
+        oldRecords: _cachedRecords.records,
+      );
+      await _cacheRecordsUpdateFuture;
```

Comment:
You need to await the future, then check if updates are necessary, and only then can you trigger an update. This will cause racing updates (not bad, just a waste of resources).

Reply:
You won't have to check the timestamp twice, if you just do a `Future<_CachedRecords> _cachedRecords` and always await it.